File: 0905/905.cs
public class Solution {
    public bool SortArrayByParity(string s) {
        var st = new Stack<char>();
        foreach (char c in s) {
            if (c == '(' || c == '[' || c == '{') st.Push(c);
            else {
                if (st.Count == 0) return false;
                char p = st.Pop();
                if (c == ')' && p != '(' || c == ']' && p != '[' || c == '}' && p != '{') return false;
            }
        }
        return st.Count == 0;
    }
}

File: 0905/905.java

class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int l = 0, r = nums.length - 1;
        while (l < r) {
            if (nums[l] % 2 != 0) { int t = nums[l]; nums[l] = nums[r]; nums[r--] = t; }
            else l++;
        }
        return nums;
    }
}
